### PR TITLE
User agent

### PR DESCRIPTION
**Pull request by jeffmendoza**
Wire up the ldflags to do something, add versions to cli commands. Set user agent on github, deps.dev, gcs, and oci access.

OSV will need: https://github.com/google/osv-scanner/pull/357

#768

### CODE REVIEW

File: pkg/handler/collector/oci/oci.go [hunk 1]
```diff
@@ -128,6 +129,7 @@ func (o *ociCollector) getTagsAndFetch(ctx context.Context, repo string, tags []
 	rcOpts := []regclient.Opt{}
 	rcOpts = append(rcOpts, regclient.WithDockerCreds())
 	rcOpts = append(rcOpts, regclient.WithDockerCerts())
+	rcOpts = append(rcOpts, regclient.WithUserAgent(version.UserAgent))
```
Comment (pxp928) [inline pkg/handler/collector/oci/oci.go:132]: nice that regclient has this!

**Review by lumjjb — APPROVED** [review]
LGTM, couple minor comments. For the OSV stuff, can you update the issue or open up another issue linking it to say that that's still missing so we don't miss it!

File: pkg/version/version.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	UserAgent    string
+	UATransport  http.RoundTripper
+)
+
+type uat struct {
+	tr http.RoundTripper
+}
+
+func init() {
+	UserAgent = fmt.Sprintf("GUAC/%s", Version)
```
[the remark this answers is not in the document]
Comment (jeffmendoza) [reply pkg/version/version.go:35]: The version looks something like this `v0.0.1-210-g7132af3` from [`git describe --tags --always`](https://git-scm.com/docs/git-describe) in the Makefile. I'm not sure if we can or will override this in the Buildback build for when we cut releases. For example, I know [goreleaser can do this](https://goreleaser.com/cookbooks/using-main.version/)

**Comment by jeffmendoza** [comment]
> For the OSV stuff, can you update the issue or open up another issue linking it to say that that's still missing so we don't miss it!

I'll update the issue with what is remaining when this merges.

File: pkg/version/version.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (lumjjb) [reply pkg/version/version.go:35]: sgtm

**Review by pxp928 — APPROVED** [review]

**Pull request merged**